modules/core/src/main/java/org/apache/ignite/internal/processors/cache/verify/IdleVerifyUtility.java

```diff
@@ -85,28 +94,117 @@ public static void checkPartitionsPageCrcSum(
         for (int pageNo = 0; pageNo < pageStore.pages(); pageId++, pageNo++) {
             buf.clear();
 
-            if (cpFlag.get())
-                throw new GridNotIdleException(CLUSTER_NOT_IDLE_MSG);
-
             pageStore.read(pageId, buf, true);
         }
+    }
+
+    /**
+     * Gather updateCounters info.
+     *
+     * @param ign Ignite instance.
+     * @param grpIds Group Id`s.
+     * @return Current group id`s and distribution partitions with update counters per group.
+     */
+    public static T2<Set<Integer>, Map<Integer, Set<Map.Entry<Integer, Long>>>> updCountersSnapshot(
```

Comment:
If I understand implementation correctly, this method either returns T2 with set containing the same elements as map.keySet or throws GridNotIdleException.
If this set is not used for some other purposes we can decrease complexity of returned data structure.

Another possible simplification: if we use map<partId, updateCounter> instead of Set<Map.Entry<partId, updateCounter> will it break anything?